src/main/scala/PropertyChecker.scala

```diff
@@ -630,4 +664,285 @@ object PropertyChecker {
+            if(matcher.matches()) {
+              var validTitles = Array[String]()
+              fieldsAndValue.getValue match {
+                case s:TextNode => validTitles = validTitles :+ s.asText()
+                case a:ArrayNode => {
+                  val titles = Array.from(a.elements().asScala)
+                  for(title <- titles) {
+                    title match {
+                      case s:TextNode => validTitles = validTitles :+ s.asText()
+                      case _ => warnings = warnings :+ PropertyChecker.invalidValueWarning
+                    }
+                  }
+                }
+                case _ => warnings = warnings :+ PropertyChecker.invalidValueWarning
+              }
+              val validTitlesArrayNode:ArrayNode = PropertyChecker.mapper.valueToTree(validTitles)
+              valueCopy.set(elementKey, validTitlesArrayNode)
+            } else {
+              valueCopy.remove(elementKey)
+              warnings = warnings :+ "invalid_language"
+            }
+          }
+          (valueCopy, warnings, csvwPropertyType)
+        }
+        case _ => (NullNode.getInstance(), Array[String](PropertyChecker.invalidValueWarning), csvwPropertyType)
+      }
+    }
+  }
+
+  def nameProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      value match {
+        case s: TextNode => {
+          val matcher = PropertyChecker.NameRegExp.pattern.matcher(s.asText())
+          if (matcher.matches()) {
+            (s, Array[String](), PropertyType.Column)
+          } else {
+            (NullNode.instance, Array[String](PropertyChecker.invalidValueWarning), PropertyType.Column)
+          }
+        }
+        case _ => (NullNode.instance, Array[String](PropertyChecker.invalidValueWarning), PropertyType.Column)
+      }
+    }
+  }
+
+  def encodingProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      value match {
+        case s: TextNode if PropertyCheckerConstants.ValidEncodings.contains(s.asText()) => {
+          (s, Array[String](), PropertyType.Dialect)
+        }
+        case _ => (NullNode.instance, Array[String](PropertyChecker.invalidValueWarning), PropertyType.Dialect)
+      }
+    }
+  }
+
+  def arrayProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      value match {
+        case a: ArrayNode => (a, Array[String](), csvwPropertyType)
+        case _ => (BooleanNode.getFalse, Array[String](PropertyChecker.invalidValueWarning), csvwPropertyType)
+      }
+    }
+  }
+
+  def trimProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
```

Comment:
Is the `csvwPropertyType` param used here?

Reply:
Fixed